File: src/commands/remote.js
import { simpleGit } from "simple-git";
import fs from "fs";

const git = simpleGit();

export default function remoteCmds(program) {
    const remoteCmd = program.command("remote").description("Manage remote repository URLs");

    remoteCmd
        .command("add <url>")
        .description("Add new origin URL")
        .action(async (url) => {
            try {
                if (!fs.existsSync(".git")) {
                    global.log.info("No git repository found. Initializing...");
                    await git.init();
                }
                await git.addRemote("origin", url);
                global.log.info(`Remote origin added successfully: ${url}`);
            } catch (error) {
                global.log.error("Failed to add remote. Make sure origin does not already exist.");
            }
        });

    remoteCmd
        .command("set <url>")
        .description("Change existing origin URL")
        .action(async (url) => {
            try {
                await git.remote(["set-url", "origin", url]);
                global.log.info(`Remote origin changed successfully to: ${url}`);
            } catch (error) {
                global.log.error("Failed to change remote. Make sure origin exists.");
            }
        });

    remoteCmd
        .command("check")
        .description("Check current remote URLs")
        .action(async () => {
            try {
                const remotes = await git.getRemotes(true);
                if (remotes.length === 0) {
                    global.log.info("No remote URL found in this project.");
                    return;
                }
                global.log.info("Remote Repositories:");
                remotes.forEach((r) => {
                    global.log.info(`   - ${r.name}: ${r.refs.push}`);
                });
            } catch (error) {
                global.log.error("Error: Not a git repository or Git not installed.");
                global.log.error(error.message);
            }
        });
}

File: cmd/remote.go
package cmd

import (
	"os"

	"github.com/ramadanny/gits/internal/gitops"
	"github.com/ramadanny/gits/internal/logger"
	"github.com/spf13/cobra"
)

var remoteCmd = &cobra.Command{
	Use:   "remote",
	Short: "Manage remote repository URLs",
}

func init() {
	rootCmd.AddCommand(remoteCmd)

	remoteCmd.AddCommand(&cobra.Command{
		Use:   "add [url]",
		Short: "Add new origin URL",
		Args:  cobra.ExactArgs(1),
		Run: func(cmd *cobra.Command, args []string) {
			if _, err := os.Stat(".git"); os.IsNotExist(err) {
				logger.Info("No git repository found. Initializing...")
				gitops.Run("init")
			}
			_, err := gitops.Run("remote", "add", "origin", args[0])
			if err != nil {
				logger.Error("Failed to add remote. Make sure origin does not already exist.")
				return
			}
			logger.Info("Remote origin added successfully: " + args[0])
		},
	})

	remoteCmd.AddCommand(&cobra.Command{
		Use:   "set [url]",
		Short: "Change existing origin URL",
		Args:  cobra.ExactArgs(1),
		Run: func(cmd *cobra.Command, args []string) {
			_, err := gitops.Run("remote", "set-url", "origin", args[0])
			if err != nil {
				logger.Error("Failed to change remote. Make sure origin exists.")
				return
			}
			logger.Info("Remote origin changed successfully to: " + args[0])
		},
	})

	remoteCmd.AddCommand(&cobra.Command{
		Use:   "check",
		Short: "Check current remote URLs",
		Run: func(cmd *cobra.Command, args []string) {
			out, err := gitops.Run("remote", "-v")
			if err != nil || out == "" {
				logger.Info("No remote URL found in this project.")
				return
			}
			logger.Info("Remote Repositories:\n" + out)
		},
	})
}